src/protocol/sort/bit_permutation.rs

```diff
@@ -109,11 +108,7 @@ mod tests {
         let h1_future = bit_permutation(ctx1, shares[1].as_slice());
         let h2_future = bit_permutation(ctx2, shares[2].as_slice());
 
-        let result: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
-            .await
-            .unwrap()
-            .try_into()
-            .unwrap();
+        let result: [_; 3] = join3(h0_future, h1_future, h2_future).await;
```

Comment:
```suggestion
        let result = join3(h0_future, h1_future, h2_future).await;
```
No need for the type annotation here.